tools/validate_reflection.py

```diff
@@ -0,0 +1,713 @@
+                debug("ignore or swap %s --> %s, ignored count=%s, swapped count=%s" % (next_reflect_class, ignore_swap, len(reflect_class.ignored), len(reflect_class.swapped)))
+                current = next_reflect_ignores_swaps.find(ignore_swap_match.group(0), current) + len(ignore_swap_match.group(0))
+            else:
+                break
+        
+
+    def add_field(self, reflect_class_name, field):
+        reflect_class = self.find_or_add(reflect_class_name)
+        assert field not in reflect_class.fields, "Reflection for %s repeats field \"%s\"" % (reflect_class_name, field)
+        reflect_class.fields.append(field)
+        debug("add_field %s --> %s" % (reflect_class_name, field))
+
+def replace_multi_line_comment(match):
+    all=match.group(1)
+    all=EmptyScope.strip_extra_pattern.sub("", all)
+    debug("multiline found=%s" % (all))
+    match=EmptyScope.ignore_swap_pattern.search(all)
+    if match:
+        ignore_or_swap = match.group(1) 
+        all = match.group(2)
+        debug("multiline %s now=%s" % (ignore_or_swap, all))
+        invalid_chars=EmptyScope.invalid_chars_pattern.search(all)
+        if invalid_chars:
+            for ic in invalid_chars.groups():
+                debug("invalid_char=%s" % (ic))
+            debug("WARNING: looks like \"%s\" is intending to %s, but there are invalid characters - \"%s\"" % (all, ignore_or_swap, ",".join(invalid_chars.groups())))
+            return ""
+        groups=re.findall(EmptyScope.multi_line_comment_ignore_swap_pattern, all)
+        if groups is None:
+            return ""
+        rtn_str="// %s " % (ignore_or_swap)
+        add=False
+        for group in groups:
+            debug("group=%s" % (group))
+            if group is None:
+                continue
+            if add:
+                rtn_str+=", "
+            rtn_str+="%s" % (group)
+            add=True
+        debug("multiline rtn_str=%s" % (rtn_str))
+        return rtn_str
+    
+    debug("multiline no match")
+    return ""
+
+def replace_line_comment(match):
+    all=match.group(0)
+    debug("singleline found=%s" % (all))
+    if EmptyScope.single_comment_ignore_swap_pattern.match(all):
+        return all
+    else:
+        return "\n"
+
+def validate_file(file):
+    f = open(file, "r")
+    contents = f.read()
+    f.close()
+    contents = "\n" + contents   # lazy fix for complex regex
+    contents = EmptyScope.multi_line_comment_pattern.sub(replace_multi_line_comment, contents)
+    contents = EmptyScope.single_comment_pattern.sub(replace_line_comment, contents)
+    found = re.search(fc_reflect_str, contents)
+    if found is None:
+        return
+    print("validate %s" % (file))
+    debug("validate %s" % (file))
+    global_namespace=Namespace("", None, 0, contents, None)
+    global_namespace.read()
+    if args.debug:
+        _, filename = os.path.split(file)
+        f = open(temp_dir + "/" + filename + ".struct", "w")
```

Comment:
Replacing this and next two lines:
```python
with open(os.path.join(temp_dir, filename + '.struct'), 'w') as f:
    f.write("global_namespace=%s" % (global_namespace))
```
`close()` becomes implicit and will be executed even if there's an exception during `write()`.

Reply:
Ok